src/aead/chacha20_poly1305.rs

```diff
@@ -131,12 +143,24 @@ fn chacha20_poly1305_open(
     in_out: &mut [u8],
     src: RangeFrom<usize>,
     cpu_features: cpu::Features,
-) -> Tag {
+) -> Result<Tag, error::Unspecified> {
     let chacha20_key = match key {
         aead::KeyInner::ChaCha20Poly1305(key) => key,
         _ => unreachable!(),
     };
 
+    let unprefixed_len = in_out
+        .len()
+        .checked_sub(src.start)
+        .ok_or(error::Unspecified)?;
+    if unprefixed_len > MAX_IN_OUT_LEN {
+        return Err(error::Unspecified);
+    }
```

Comment:
This code path only occurs with a (practically) impossibly-long input, so we can't test it.